src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -125,20 +124,55 @@ private void AddRuleset()
+            //
+            // B) Old-style project, file is *not* ref'd, but SonarLint.xml file exists on disc in the same folder as the project file.
+            //      We detect the project is not fully bound -> try to add the file.
+            //      This will error with:
+            //          "Cannot add a link to the file SonarLint.xml. There is already a file of the same name in this folder.".
+            //      This should be a real corner case.
+            // 
+            // C) New-style project, file is ref'd in a Directory.Build.props file but with the wrong ItemType
+            //      We detect the project is not fully bound -> try to change the item type.
+            //      This errors with:
+            //          "Only included item inside the project can be moved.
+            //           Parameter name: currentItem"
+            //      Note: this only happens if the ItemType is one that VS knows e.g. "Content". If it is a type
+            //      VS does not recognise (e.g. "Foo") then adding the file succeeds.
+
+            var additionalFilePath = cSharpVBBindingConfig.AdditionalFile.Path;
+            var refdFile = projectSystem.FindFileInProject(initializedProject, additionalFilePath);
+            if (refdFile == null)
             {
-                var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
-                projectSystem.AssertLocalServiceIsNotNull();
-                // It's possible that the additional file is set property, but the ruleset is out of date, which will trigger the binding. So we need to remove the file first, otherwise we get an error that the file already exists.
-                // It's possible that the additional file exists but has a wrong type, e.g. Compile/Content, and not AdditionalFiles. In which case removing and re-adding it solves the problem.
-                projectSystem.RemoveFileFromProject(initializedProject, cSharpVBBindingConfig.AdditionalFile.Path);
-                projectSystem.AddFileToProject(initializedProject, cSharpVBBindingConfig.AdditionalFile.Path, "AdditionalFiles");
+                projectSystem.AddFileToProject(initializedProject, additionalFilePath, Constants.AdditionalFilesItemTypeName);
             }
-            catch (Exception ex)
+            else
             {
-                throw new Exception($"Failed to bind project {initializedProject.FullName}, exception details: {ex}");
+                var property = VsShellUtils.FindProperty(refdFile.Properties, Constants.ItemTypePropertyKey);
+                if (!Constants.AdditionalFilesItemTypeName.Equals(property.Value?.ToString(), StringComparison.OrdinalIgnoreCase))
+                {
+                    // This will fail if the existing item ref isn't in the main project file e.g. if it is in a Directory.Build.props file.
+                    // We can't do much about this, other than catch and log the error higher up.
```

Comment:
The binding will fail with the rather cryptic `Only included item inside the project can be moved.` mentioned above.
We can discuss whether it's worth logging more useful error here.
However, it should be very uncommon - the correct file needs to be referenced in a `Directory.Build.props`, with an item type that VS recognises.

Adding the exception handling and logging is straightforward, but it would require a re-write of the tests to inject a failure at the right point.